ddraw/IDirect3DDeviceX.cpp

```diff
@@ -1431,28 +1458,67 @@ HRESULT m_IDirect3DDeviceX::BeginScene()
 		// Set 3D Enabled
 		ddrawParent->Enable3D();
 
-		return (*d3d9Device)->BeginScene();
+		hr = (*d3d9Device)->BeginScene();
 	}
 
 	switch (ProxyDirectXVersion)
 	{
 	case 1:
-		return GetProxyInterfaceV1()->BeginScene();
+		hr = GetProxyInterfaceV1()->BeginScene();
+		break;
+
 	case 2:
-		return GetProxyInterfaceV2()->BeginScene();
+		hr = GetProxyInterfaceV2()->BeginScene();
+		break;
+
 	case 3:
-		return GetProxyInterfaceV3()->BeginScene();
+		hr = GetProxyInterfaceV3()->BeginScene();
+		break;
+
 	case 7:
-		return GetProxyInterfaceV7()->BeginScene();
-	default:
-		return DDERR_GENERIC;
+		hr = GetProxyInterfaceV7()->BeginScene();
+		break;
+	}
+
+#if WITH_IMGUI
+	if(SUCCEEDED(hr) || Config.Dd7to9)
+	{
+		ImGui_ImplDX9_NewFrame();
+		ImGui_ImplWin32_NewFrame();
+		ImGui::NewFrame();
 	}
+#endif
+
+	return hr;
 }
 
 HRESULT m_IDirect3DDeviceX::EndScene()
 {
 	Logging::LogDebug() << __FUNCTION__ << " (" << this << ")";
 
+#if WITH_IMGUI
+	static bool ShowDebugUI = false;
+	if (ImGui::IsKeyDown(ImGui::GetKeyIndex(ImGuiKey_LeftAlt)) &&
+		ImGui::IsKeyPressed(ImGui::GetKeyIndex(ImGuiKey_D), false))
+	{
+		ShowDebugUI = !ShowDebugUI;
+	}
+
+	if(ShowDebugUI)
+	{
+		ImGui::Begin("Hello, world!");
+		ImGui::Text("This is some text.");
+		ImGui::End();
+
+		ImGui::Render();
+		ImGui_ImplDX9_RenderDrawData(ImGui::GetDrawData());
+	}
+	else
+	{
+		ImGui::EndFrame();
+	}
+#endif
```

Comment:
Again, this section should probably be inside the `Config.Dd7to9` section.  This also causes a crash if `EnableDdrawWrapper` is enabled but not `Dd7to9`.